2023-08-30 14:52:02,509 - worker_process - INFO - Worker_process started.
2023-08-30 14:52:02,751 - Cell - INFO - site-1.48f8c190-3a43-4423-a8b8-94437dcbd9b9: created backbone internal connector to tcp://localhost:1078 on parent
2023-08-30 14:52:02,774 - Cell - INFO - site-1.48f8c190-3a43-4423-a8b8-94437dcbd9b9: created backbone external connector to grpc://192.168.100.3:8102
2023-08-30 14:52:02,774 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00001 ACTIVE tcp://localhost:1078] is starting
2023-08-30 14:52:02,774 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00002 ACTIVE grpc://192.168.100.3:8102] is starting
2023-08-30 14:52:02,775 - FederatedClient - INFO - Wait for client_runner to be created.
2023-08-30 14:52:02,775 - FederatedClient - INFO - Got client_runner after 0.00021600723266601562 seconds
2023-08-30 14:52:02,775 - FederatedClient - INFO - Got the new primary SP: grpc://192.168.100.3:8102
2023-08-30 14:52:02,787 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: client runner started
2023-08-30 14:52:02,810 - Communicator - INFO - Received from secure_project server  (1114733 Bytes). getTask: train time: 0.015916824340820312 seconds
2023-08-30 14:52:02,811 - FederatedClient - INFO - pull_task completed. Task name:train Status:True 
2023-08-30 14:52:02,811 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: got task assignment: name=train, id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d
2023-08-30 14:52:02,812 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, task_name=train, task_id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d]: invoking task executor LearnerExecutor
2023-08-30 14:52:02,812 - LearnerExecutor - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, task_name=train, task_id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d]: Client trainer got task: train
2023-08-30 14:52:02,812 - MNISTLearner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, task_name=train, task_id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d]: Client site-1 initialized at 
 /home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../48f8c190-3a43-4423-a8b8-94437dcbd9b9/app_site-1 
 with args: Namespace(workspace='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/..', startup='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../startup', token='46f86140-8de3-4521-841f-6972de9e2a26', ssid='ebc6125d-0a56-4688-9b08-355fe9e4d61a', job_id='48f8c190-3a43-4423-a8b8-94437dcbd9b9', client_name='site-1', sp_target='192.168.100.3:8102', parent_url='tcp://localhost:1078', fed_client='fed_client.json', set=['secure_train=true', 'uid=site-1', 'org=nvidia', 'config_folder=config', 'print_conf=True'], local_rank=0, train_config='config/config_train.json', client_config='config/config_fed_client.json', config_folder='config', env='config/environment.json')
2023-08-30 14:52:04,137 - ClientRunner - ERROR - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, task_name=train, task_id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d]: Processing error from executor LearnerExecutor: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-08-30 14:52:04,138 - ClientRunner - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/private/fed/client/client_runner.py", line 254, in _process_task
    reply = executor.execute(task.name, task.data, fl_ctx, self.task_abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 86, in execute
    return self.train(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 99, in train
    validate_result: Shareable = self.learner.validate(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 357, in validate
    self._create_datasets(fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 165, in _create_datasets
    self.train_loader = self.train_datamgr.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers, train=True)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 39, in get_loader
    return data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-08-30 14:52:04,139 - FederatedClient - INFO - Starting to push execute result.
2023-08-30 14:52:04,143 - Communicator - INFO -  SubmitUpdate size: 575 Bytes. time: 0.003359079360961914 seconds
2023-08-30 14:52:04,143 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, task_name=train, task_id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d]: result sent to server for task: name=train, id=41a9be44-b5e5-45e4-85e8-6bfdde5c620d
2023-08-30 14:52:04,985 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9, peer=secure_project, peer_run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: received aux request from Server to end current RUN
2023-08-30 14:52:04,986 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: ABORT (RUN) command received
2023-08-30 14:52:04,986 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: ABORT (RUN) requests end run events sequence
2023-08-30 14:52:04,987 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: ABOUT_TO_END_RUN fired
2023-08-30 14:52:04,988 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: END_RUN fired
2023-08-30 14:52:06,146 - ClientRunner - INFO - [identity=site-1, run=48f8c190-3a43-4423-a8b8-94437dcbd9b9]: run method requests end run events sequence
2023-08-30 14:52:06,161 - FederatedClient - INFO - Shutting down client run: site-1
2023-08-30 14:52:08,116 - MPM - INFO - MPM: Good Bye!
